### PR TITLE
Maths Module Restructured

### LINKED ISSUE #14: Implement Basic Math Algorithms
Currently, python provides basic math operations but they are not that flexible. Like you can calculate H.C.F.  of two numbers but what if we want to calculate H.C.F. of a list. 
Also, we can implement other algorithms like for prime, next prime, last prime etc.

### PR DESCRIPTION
**Pull request by ravi5175**
# Changes
- restructured maths module from scratch to make it usable for dsalgo package
- removed need for external dependency
- added new algorithms
image

- added unit test

### Related Issue
Fixes #14

### Type of Change:
- Code
- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- Bugs fixed
- This change requires a documentation update

### How Has This Been Tested?

- unittest passed
image

### Checklist:

- [x] I read the **CONTRIBUTING** document before making this pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] My change requires change to documentation
- [x] I have updated documentation accordingly
- [ ] I have updated the requirements file
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: tests/dmath_test.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        #case 3 start == 31 end == none reverse == true count == default
+        self.assertEqual(self.dm.series_prime(start=31,reverse=True),[31,29,23,19,17,13,11,7,5,3]) 
+
+        #case 4 start == 31 end == 2 reverse == true count = 5
+        self.assertEqual(self.dm.series_prime(start=31,reverse=True,count=5),[31,29,23,19,17]) 
+
+        
+if __name__ == '__main__':
+    unittest.main()
```
Comment (codesankalp) [inline tests/dmath_test.py:63]: no new line at the end of file

File: dsalgo/math.py [hunk 2]
```diff
@@ -12,186 +12,198 @@
 #hcf_k_elements(l,k) - HCF of k elements in a list
 #lcm_k_elements(l, k) - LCM of k elements in a list
 
+class dmath():
```
Comment (codesankalp) [inline dsalgo/math.py:15]: reason for naming it dmath?

**Review by codesankalp — COMMENTED** [review]
Fine by me.

File: dsalgo/math.py (hunk 2 above)
Comment (devkapilbansal) [reply dsalgo/math.py:15]: @ravi5175 please use function_names that explains what that function is for

Comment (ravi5175) [reply dsalgo/math.py:15]: wait there's something wrong,let me correct

Comment (ravi5175) [reply dsalgo/math.py:15]: @codesankalp reason for using dmath is reduce confusion among actual python math module and dsalgo math module.

Comment (ravi5175) [reply dsalgo/math.py:15]: @devkapilbansal there was some branching issue thats why my file was not updated and function name with discription was not present, i have solved the issue, function description is now available with individual doc strings added to each functions

File: tests/dmath_test.py (hunk 1 above)
Comment (ravi5175) [reply tests/dmath_test.py]: solved

**Comment by ravi5175** [comment]
@devkapilbansal  can you clarify about changes you requested??

**Comment by devkapilbansal** [comment]
Pinging @17sushmita @keshavgbpecdelhi for review

File: tests/dmath_test.py (hunk 1 above)
Comment (17sushmita) [inline tests/dmath_test.py]: No need of this driver code

File: tests/dmath_test.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import unittest
```
Comment (17sushmita) [inline tests/dmath_test.py:1]: rename this file to test_dmath.py

File: dsalgo/dmath.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+            
+                
+
+
+            
+
+                    
+        
+
```
Comment (17sushmita) [inline dsalgo/dmath.py:176]: No need of these blank lines

**Review by 17sushmita — CHANGES_REQUESTED** [review]

**Comment by ravi5175** [comment]
@17sushmita requested changes are done, please find more pins from the haystack

File: dsalgo/dmath.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    series_prime(self,start = 2,count=10,reverse=False,end=None) - returns \
+    list type array of prime as per arguments
+    '''
+
+    def gcd(self, a, b):
+        '''
+            args: int a ,int b
+            logic: eucledians gcd
+            returns: gcd of two integers
+
```
Comment (devkapilbansal) [inline dsalgo/dmath.py:22]: Why a blank line here??

File: tests/test_dmath.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                         [2, 3, 5, 7, 11])
+
+        # case 3 start == 31 end == none reverse == true count == default
+        self.assertEqual(self.dm.series_prime(start=31, reverse=True),
+                         [31, 29, 23, 19, 17, 13, 11, 7, 5, 3])
+
+        # case 4 start == 31 end == 2 reverse == true count = 5
+        self.assertEqual(self.dm.series_prime(start=31, reverse=True, count=5),
+                         [31, 29, 23, 19, 17])
+
```
Comment (devkapilbansal) [inline tests/test_dmath.py:63]: Add a new line here

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@ravi5175 please add appropriate messages in your tests also. So that it is easy to understand which test case is running and when

File: tests/test_dmath.py (hunk 6 above)
Comment (ravi5175) [reply tests/test_dmath.py:63]: why

**Comment by devkapilbansal** [comment]
Also, rebase your branch with current master

**Comment by devkapilbansal** [comment]
@ravi5175 are you working on this?

**Comment by ravi5175** [comment]
@devkapilbansal yes i am working on it

**Comment by codesankalp** [comment]
image

Tested by this and manually also.

merging after rebasing and squashing.

**Pull request merged**